source/fab/steps/archive_objects.py

```diff
@@ -49,46 +51,54 @@ def archive_objects(config: BuildConfig,
     * The object archive is a convenience step before linking a **shared object**.
     * One or more object archives as convenience steps before linking **executables**.
 
-    The benefit of creating an object archive before linking is simply to reduce the size
-    of the linker command, which might otherwise include thousands of .o files, making any error output
-    difficult to read. You don't have to use this step before linking.
-    The linker step has a default artefact getter which will work with or without this preceding step.
+    The benefit of creating an object archive before linking is simply to
+    reduce the size of the linker command, which might otherwise include
+    thousands of .o files, making any error output difficult to read. You
+    don't have to use this step before linking. The linker step has a default
+    artefact getter which will work with or without this preceding step.
 
     **Creating a Static or Shared Library:**
 
-    When building a library there is expected to be a single build target with a `None` name.
-    This typically happens when configuring the :class:`~fab.steps.analyser.Analyser` step *without* a root symbol.
-    We can assume the list of object files is the entire project source, compiled.
+    When building a library there is expected to be a single build target
+    with a `None` name. This typically happens when configuring the
+    :class:`~fab.steps.analyser.Analyser` step *without* a root symbol.
+    We can assume the list of object files is the entire project source,
+    compiled.
```

Comment:
This is true for the moment but I have a plan to use source directives to mark up the API which would allow "zero configuration" mode to produce multiple libraries. I suppose it's "implied configuration mode" at that point.

Reply:
"Mark up" as in special comments/directives in the source files? Interesting idea. For most Fab use cases that I can think of, zero-config won't really ever be used, so I am kind of neutral here - if LFRic/um/... needs libs, that can be done (with some extensions to Fab I think??) in a Fab script.